spyder/widgets/editor.py

```diff
@@ -2454,18 +2510,44 @@ def split(self, orientation=Qt.Vertical):
             current_editor.setFocus()
 
     def iter_editorstacks(self):
+        """Return the editor stacks for this splitter and every first child.
+
+        Note: If a splitter contains more than one splitter as a direct
+              child, only the first child's editor stack is included.
+
+        Returns:
+            List of tuples containing (EditorStack instance, orientation).
+        """
         editorstacks = [(self.widget(0), self.orientation())]
         if self.count() > 1:
             editorsplitter = self.widget(1)
             editorstacks += editorsplitter.iter_editorstacks()
         return editorstacks
 
     def get_layout_settings(self):
-        """Return layout state"""
+        """Return the layout state for this splitter and its children.
+
+        Record the current state, including file names and current line
+        numbers, of the splitter panels.
+
+        Returns:
+            A dictionary containing keys {hexstate, sizes, splitsettings}.
+                hexstate: String of saveState() for self.
+                sizes: List for size() for self.
+                splitsettings: List of tuples of the form
+                       (orientation, cfname, clines) for each EditorSplitter
+                       and its EditorStack.
+                           orientation: orientation() for the editor
+                                 splitter (which may be a child of self).
+                           cfname: EditorStack current file name.
+                           clines: Current line number for each file in the
+                               EditorStack.
+        """
         splitsettings = []
         for editorstack, orientation in self.iter_editorstacks():
             clines = []
             cfname = ''
+            # XXX - this overrides value from the loop to always be False?
             orientation = False

```

Comment:
I think you are right with the value for `orientation`, but I'm not quite sure. @ccordoba12 do you know the reason to do this? After clarifing this, please @csabella, write the explanation instead of the current comment that you left.

Reply:
@dalthviz   This was changed recently by andfoy and I put this comment so that it could be researched further.  What I found so far is that the `orientation` boolean from here is not used in any of the loops that read `splitsettings` (in plugins/editor or widgets/editor) except for in `set_layout_settings` in `EditorSplitter` which has the following line (it is mapped into `is_vertical` in the loop):

    splitter.split(Qt.Vertical if is_vertical else Qt.Horizontal)

but somehow the orientation of the splits end up correct on the restore.  I believe that `hexstate` within `set_layout_settings` is applying the correct split orientation after the fact, so `orientation` may not even need to be saved at all in `splitsettings`.  Although, after Travis issues, I'm not sure what PyQt4 is doing, but it's definitely a different hexstate than PyQt5. 

Interestingly, in `set_layout_settings`, even `cfname` isn't used.  There is an inline comment about `go_to_line` and I had quite a bit of difficulty writing a test for the line search within `set_layout_settings` because it just seemed to not fit in with the rest of the code.

Reply:
@andfoy, please chime in this discussion. I haven't touched the splitting code, so I can't say anything constructive here, sorry.

Reply:
@ccordoba12 @csabella To be honest, last time that I modified the splitting behaviour, I was not sure about some of the settings and definitions that were disposed there.

Reply:
It was changed in https://github.com/spyder-ide/spyder/pull/4869, but personally I didn't find a good reason for setting it to `False`, although if it's working is better not to change it (and debbuging it is out of the scope of this PR)

Reply:
Ok, thanks @rlaverde. Let's merge this one as it is.

We'll see later how to improve this logic.